tests_common/pytest_plugin.py

```diff
@@ -50,6 +50,23 @@
 
     Op = TypeVar("Op", bound=BaseOperator)
 
+# NOTE: DO NOT IMPORT AIRFLOW THINGS HERE!
+#
+# This plugin is responsible for configuring Airflow correctly to run tests.
+# Importing Airflow here loads Airflow too eagerly and break the configurations.
+# Instead, import what you want lazily inside a fixture function.
+#
+# Be aware that many things in tests_common also indirectly imports Airflow, so
+# those modules also should not be imported globally.
+#
+# (Things in the TYPE_CHECKING block are fine because they are not actually
+# imported at runtime; those imports are only hints to the type checker.)
+
+assert "airflow" not in sys.modules, (
+    "Airflow SHOULD NOT have been imported at this point! "
+    "Read comments in pytest_plugin.py to understand more."
```

Comment:
nit: Is it worth changing to `Path(__file__).absolute()` in case this file gets moe around in future?

Reply:
I trust future contributors to at least lightly browse through the file when they move it… (practically this probably won’t be moved that much)

Putting an absolute path here seems a bit too excessive to me

Reply:
🤞